OpenSim/Simulation/Test/testDynamicsParallelism.cpp

```diff
@@ -0,0 +1,117 @@
+    return 0;
+}
+
+//==============================================================================
+// Test Cases
+//==============================================================================
+void testDynamicsParallelism(const string& modelFile)
+{
+    using namespace SimTK;
+    using namespace std;
+
+    //==========================================================================
+    // Setup OpenSim model
+    Model model(modelFile);
+
+    ControlSetController* controller = new ControlSetController();
+    controller->setControlSetFileName("arm26_StaticOptimization_controls.xml");
+    model.addController(controller);
+
+    //==========================================================================
+    // Test Accuracy of Dynamics
+    
+    //Realize to Stage::Dynamics once to obtain a Dyanmics "answer"
+    State& state = model.initSystem();
+    model.getSystem().realize(state, Stage::Dynamics);
+    const MultibodySystem& mbs = model.getMultibodySystem();
+    Vector_<SpatialVec>&   rigidBodyForcesAnswer = mbs.updRigidBodyForces(state,
+                                                               Stage::Dynamics);
+    string rigidBodyForcesAnswerString = rigidBodyForcesAnswer.toString();
+    Vector_<Vec3>&         particleForcesAnswer = mbs.updParticleForces(state,
+                                                               Stage::Dynamics);
+    string particleForcesAnswerString = particleForcesAnswer.toString();
+    Vector&                mobilityForcesAnswer = mbs.updMobilityForces(state,
+                                                               Stage::Dynamics);
+    string mobilityForcesAnswerString = mobilityForcesAnswer.toString();
+
+    //Constantly realize to Stage::Dynamics to make sure that our answers are
+    //consistent
+    for(int x = 0; x < 100; ++x)
+    {
+      State& state = model.initSystem();
+
+      model.getSystem().realize(state, Stage::Dynamics);
+      const MultibodySystem& mbs = model.getMultibodySystem();
+      Vector_<SpatialVec>   tempRigidBodyForces = mbs.updRigidBodyForces(state,
+                                                               Stage::Dynamics);
+      Vector_<Vec3>&    tempParticleForcesAnswer = mbs.updParticleForces(state,
+                                                               Stage::Dynamics);
+      Vector&            tempMobilityForcesAnswer = mbs.updMobilityForces(state,
+                                                               Stage::Dynamics);
+                                                               
+      SimTK_TEST((int)(rigidBodyForcesAnswerString ==
+                                          tempRigidBodyForces.toString()) == 1);
+      SimTK_TEST((int)(particleForcesAnswerString ==
+                                     tempParticleForcesAnswer.toString()) == 1);
+      SimTK_TEST((int)(particleForcesAnswerString ==
+                                     tempParticleForcesAnswer.toString()) == 1);
```

Comment:
> Why not use the SimTK_TEST_EQ macros?

That is for numerical tests that need a tolerance. Here is the documentation for that one:
```
/// Test that two numerical values are equal to within a reasonable numerical
/// error tolerance, using a relative and absolute error tolerance. In the
/// case of composite types, the test is performed elementwise.
```